src/lib/battery/battery.cpp

```diff
@@ -186,7 +186,8 @@ Battery::updateBatteryStatus(hrt_abstime timestamp, float voltage_v, float curre
 void
 Battery::publish()
 {
-	orb_publish_auto(ORB_ID(battery_status), &_orb_advert, &_battery_status, &_orb_instance);
+	int dummy; // We're not interested in the instance ID we get
+	orb_publish_auto(ORB_ID(battery_status), &_battery_status_pub, &_battery_status, &dummy);
```

Comment:
If you're going to change this how about replacing it with a `uORB::PublicationMulti<battery_status_s>`?

Reply:
I thought about that but don't know what's the equivalent of this auto publication. Don't you have to specify the ID in advance? Would you just use the `const int _index;` of the battery for that?

Reply:
I tried to add it. Works with one battery (SITL). Is that the correct use? https://github.com/PX4/Firmware/pull/15616/commits/a153aa5aa66300427e06c6fd2dcd9f3ab5e34f74
Does it automatically choose the next free instance?